### ISSUE TITLE
[USABILITY] Notification Logic for Group Replies

### ISSUE THREAD

**Issue by vburgett** [rokwire/illinois-app#2938]
When a user replies to a post in a Group, the original poster plus ALL members of that Group get a notification (under the bell icon and/or a push notification). 

I would expect ONLY the original poster to get notified. 

For example, in the attached screenshot of the Group Gus, I would expect only Zia to be notified of Fatimah's response. Instead, all 135 members received the notification. 

iOS 16.0.2
iPhone 13 Pro
Version 4.2.47

![image](https://user-images.githubusercontent.com/1044905/214462271-2163a073-86ae-486b-9dd8-78a409bdefef.png)

cc: @kpecknold

**Comment by dobromirdobrev** [comment]
The notifications to group members are not manipulated in the UIUC app but in the relevant Groups BB. So a new issue is created for handling that request in that BB: https://github.com/rokwire/groups-building-block/issues/372
Once it is done, we can verify it using the UIUC app.

**Comment by vburgett** [comment]
Is this update still in progress, @dobromirdobrev? Will it align with the 5.1 release of the Illinois app?

**Comment by vburgett** [comment]
@dobromirdobrev I have updated this ticket's original request to include anyone in the post thread receiving a notification. Can you update the related Groups building block ticket as well? Thank you.

**Comment by dobromirdobrev** [comment]
Hi @vburgett ,

The backend issue https://github.com/rokwire/groups-building-block/issues/372 is closed which means it should be fixed.

Please verify it on dev when you have a chance.

**Comment by vburgett** [comment]
We tested this in 6.0.14 with a post in a Group. There are two issues:
1. When a person replies to their own post, he or she should not get notified. Currently they are. (Jake posts and then replies to the thread, he should not get a notification about his own reply.)
2. When people reply to the post, the original poster does get notified as expected but the other people in the thread do not get notified and they should.

**Comment by vburgett** [comment]
This is still an issue in 6.1.64 as described in the last post from me. @dobromirdobrev

**Comment by dobromirdobrev** [comment]
@vburgett , 

This is a backend issue - https://github.com/rokwire/social-building-block/issues/143 . We will let you know when it gets fixed.

**Comment by dobromirdobrev** [comment]
> This is still an issue in 6.1.64. [@dobromirdobrev](https://github.com/dobromirdobrev)
> 
> 1. (same as above) When a person replies to their own post, he or she should not get notified. Currently they are. (Jake posts and then replies to the thread, he should NOT get a notification about his own reply.)
> 2. When people reply to the post, the original poster does NOT get notified as expected.*
> 3. Other people who have replied to a post do NOT get notified of a new reply and they should. (Jake posts and then I reply. If Jake replies too, I should get notified about his reply.)*
> 
> *most important fixes


@vburgett ,

This should be fixed. The backend issue is closed. I checked it on my end all works as expected (all 3 points). 

Please, verify it on your end and let us know. Thanks.
